src/ideal/ideal.jl

```diff
@@ -541,25 +556,37 @@ ideal will have the same number of generators as $I$, even if they are zero.
 """
 function reduce(I::sideal{S}, G::sideal{S}) where S <: SPolyUnion
    check_parent(I, G)
+   if S <: slpalg
+      I.isTwoSided && G.isTwoSided || error("letterplace reduce requires two-sided ideals")
+   elseif S <: spluralg
+      # TODO: How does this work when base_ring(I) is a quotient from
+      #       a (necessarily) two-sided ideal?
+      !I.isTwoSided && !G.isTwoSided || error("plural reduce requires left ideals")
```

Comment:
@hannes14 Help need here. Is this the proper check here? Require both inputs to be left? If yes, then how does one check equality of two-sided Plural ideals?